`parseWithZod` fails to parse Arrays containing empty strings (`''`) correctly

### Describe the bug and the expected behavior

When an array in formData contains an empty string (`''`) as the first element, it gets trimmed, and the resulting payload has fewer elements than expected. As a result, the indices in parseResult.value do not match the original formData.
This behavior also affects validation errors when using useForm, making it difficult to pinpoint which input has an error.

### Conform version

v1.1.5

### Steps to Reproduce the Bug or Issue

```typescript
test("parseWithZod with array", () => {
  const formData = new FormData();

  formData.append("animals", "");
  formData.append("animals", "dog");
  formData.append("animals", "");
  formData.append("animals", "cat");

  const result = parseWithZod(formData, {
    schema: z.object({
      animals: z.array(z.string().optional()),
    }),
  });

  expect(result.status).toBe("success");
  expect(result.payload).toEqual({
    animals: ["", "dog", "", "cat"],
  });
});
```

```
AssertionError: expected { animals: [ 'dog', '', 'cat' ] } to deeply equal { animals: [ '', 'dog', '', 'cat' ] }

- Expected
+ Received

  Object {
    "animals": Array [
-     "",
      "dog",
      "",
      "cat",
    ],
  }
```

### What browsers are you seeing the problem on?

Chrome

### Screenshots or Videos

_No response_

### Additional context

This issue occurs because `''` is treated as falsy. As a result, setting values into the context is skipped until a concrete value is provided.

https://github.com/edmundhung/conform/blob/b0fb91dbab474c7a150d5f56cb71dbb92888779c/packages/conform-dom/submission.ts#L103-L104

--- a/.changeset/fix-array-empty-string.md
+++ b/.changeset/fix-array-empty-string.md
@@ -1,0 +1,5 @@
+---
+'@conform-to/dom': patch
+---
+
+Preserve empty string values inside array fields when building submission payloads.

--- a/packages/conform-dom/submission.ts
+++ b/packages/conform-dom/submission.ts
@@ -101,7 +101,7 @@ export function getSubmissionContext(
 			context.payload,
 			name,
 			(prev: unknown) => {
-				if (!prev) {
+				if (prev === undefined) {
 					return next;
 				} else if (Array.isArray(prev)) {
 					return prev.concat(next);

--- a/packages/conform-dom/tests/submission.test.ts
+++ b/packages/conform-dom/tests/submission.test.ts
@@ -281,3 +281,29 @@ test('parse()', () => {
 		},
 	});
 });
+
+test('parse() preserves empty strings in array fields', () => {
+	const submission = parse(
+		createFormData([
+			['animals', ''],
+			['animals', 'dog'],
+			['animals', ''],
+			['animals', 'cat'],
+		]),
+		{
+			resolve(payload) {
+				return { value: payload };
+			},
+		},
+	);
+
+	expect(submission.payload).toEqual({
+		animals: ['', 'dog', '', 'cat'],
+	});
+	expect(submission).toMatchObject({
+		status: 'success',
+		value: {
+			animals: ['', 'dog', '', 'cat'],
+		},
+	});
+});

